nodeadm/test/e2e/cases/kubelet-flags/config.yaml

```diff
@@ -0,0 +1,14 @@
+---
+apiVersion: node.eks.aws/v1alpha1
+kind: NodeConfig
+metadata:
+  name: example
+spec:
+  cluster:
+    name: my-cluster
+    apiServerEndpoint: https://example.com
+    certificateAuthority: Y2VydGlmaWNhdGVBdXRob3JpdHk=
+    cidr: 10.100.0.0/16
+  kubelet:
+    flags:
+      verbosity: '5'
```

Comment:
This usage looks like of odd. Why not use the container args approach:
```
flags:
  - -v=7
  - --cloud-provider=external
  - --image-credential-provider-config=/foo/bar
```

Reply:
The only real difference is that you are going to see all of the argument that nodeadm writes as well as the user provided arguments, when with the map you are actually overriding the value in-place; if users are more familiar with this approach then it makes sense to me. There shouldn't be a functional difference as long as the flags are appended to the end.

Reply:
The k8s components use spf13/pflag under the covers: https://github.com/kubernetes/component-base/blob/2065849162a4fdce3f03b9e255475ef0ef04390d/cli/flag/flags.go#L23

Which has the behavior that if a flag which only supports a single value is defined twice, the later value is used. So as long as what the user defines here follows our defaults (which is the case on AL2), it'll work as expected as far as overrides go.

Reply:
done, makes sense to use the typical command args approach